pkgs/by-name/st/sta/package.nix

```diff
@@ -16,8 +19,40 @@ stdenv.mkDerivation {
     sha256 = "sha256-AiygCfBze7J1Emy6mc27Dim34eLR7VId9wodUZapIL4=";
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [ autoreconfHook ];
 
+  doCheck = true;
+  nativeCheckInputs = [ cxxtest ];
+  checkInputs = [ cxxtest ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    pushd test
+
+    cxxtestgen --error-printer --have-std -o tests.cpp sta_test_1.h sta_test_2.h
+    ${stdenv.cc.targetPrefix}c++ -o tester tests.cpp
+    ./tester
+
+    popd
+
+    runHook postCheck
+  '';
+
+  passthru.tests.sanity-check = runCommand "sta-sanity-check" { nativeBuildInputs = [ sta ]; } ''
+    # Only integers to avoid weird floating point math differences between platforms
+    res=$(printf "1\n1\n1" | sta | sed -n 2p)
+    expected=$(printf "3\t1\t1\t3\t1\t0\t0\t")
+    if [ "$res" != "$expected" ]; then
+      echo "error: got '$res' as output, expected '$expected'"
+      exit 1
+    fi
+
+    touch "$out"
+  '';
```

Comment:
Could probably just remove this now that we're running the proper tests. WDYT?

Reply:
Yeah, I don't think it's necessary